cmd/sfncli/runner.go

```diff
@@ -20,36 +20,48 @@ const maxTaskOutputLength = 32768
 const maxTaskFailureCauseLength = 32768
 
 type TaskRunner struct {
-	sfnapi    sfniface.SFNAPI
-	taskToken string
-	cmd       string
-	args      []string
+	sfnapi        sfniface.SFNAPI
+	taskToken     string
+	cmd           string
+	args          []string
+	executionName string
 }
 
-func NewTaskRunner(cmd string, args []string, sfnapi sfniface.SFNAPI, taskInput string, taskToken string) TaskRunner {
-	// if the task input is an array of strings, interpret these as an args array
-	// otherwise pass the raw input as a single arg
-	var taskInputArgs []string
-	if err := json.Unmarshal([]byte(taskInput), &taskInputArgs); err != nil {
-		taskInputArgs = []string{taskInput}
+func NewTaskRunner(
+	cmd string, args []string, sfnapi sfniface.SFNAPI, input, taskToken string,
+) (TaskRunner, error) {
+	var job map[string]interface{}
+	if err := json.Unmarshal([]byte(input), &job); err != nil {
+		return TaskRunner{}, fmt.Errorf("Input must be a json object: %s", err)
 	}
 
-	return TaskRunner{
-		sfnapi:    sfnapi,
-		taskToken: taskToken,
-		cmd:       cmd,
-		args:      append(args, taskInputArgs...),
+	executionName, ok := job["$execution-name"].(string)
+	if !ok {
+		return TaskRunner{}, fmt.Errorf("Job with unknown execution-name")
```

Comment:
can this not be a terminal error? Making this terminal makes `sfncli` harder to use as an open source repo, since it requires users to follow our convention and their own way of injecting this into the input of their executions

Reply:
Removed in 83ee42e